drop-boxes/register-nanopore-dropbox/register-nanopore.py

```diff
@@ -0,0 +1,220 @@
+alt_pattern = re.compile('Q\w{4}ENTITY-[1-9][0-9]*')
+
+NANOPORE_EXP_TYPE_CODE = "Q_NGS_NANOPORE_RUN"
+NANOPORE_SAMPLE_TYPE_CODE = "Q_NGS_NANOPORE_SINGLE_SAMPLE_RUN"
+NANOPORE_LOG_CODE = "Q_NGS_NANOPORE_RUN_LOGS"
+NANOPORE_FASTQ_CODE = "Q_NGS_NANOPORE_RUN_FASTQ"
+NANOPORE_FAST5_CODE = "Q_NGS_NANOPORE_RUN_FAST5"
+NANOPORE_SAMPLE_PREFIX = "NGS"
+
+# needed for pooled samples with multiple measurements
+usedSampleIdentifiers = set()
+usedExperimentIdentifiers = set()
+
+def createNewSample(transaction, space, parentSampleCode):
+    run = 0
+    sampleExists = True
+    newSampleID = None
+    while sampleExists:
+        run += 1
+        newSampleID = '/' + space + '/' + NANOPORE_SAMPLE_PREFIX + str(run) + parentSampleCode
+        sampleExists = transaction.getSampleForUpdate(newSampleID) or newSampleID in usedSampleIdentifiers
+    usedSampleIdentifiers.add(newSampleID)
+    return transaction.createNewSample(newSampleID, NANOPORE_SAMPLE_TYPE_CODE)
+
+def createNewExperiment(transaction, space, project):
+    search_service = transaction.getSearchService()
+    existingExperiments = search_service.listExperiments("/" + space + "/" + project)
+    for eexp in existingExperiments:
+        usedExperimentIdentifiers.add(eexp.getExperimentIdentifier())
+    run = len(usedExperimentIdentifiers)
+    expExists = True
+    newExpID = None
+    while expExists:
+        run += 1
+        newExpID = '/' + space + '/' +project+ '/' + project+str(run)
+        expExists = newExpID in usedExperimentIdentifiers
+    usedExperimentIdentifiers.add(newExpID)
+    return transaction.createNewExperiment(newExpID, NANOPORE_EXP_TYPE_CODE)
+
+# return first line of file
+def getDatahandlerMetadata(incomingPath, fileName):
+    path = os.path.realpath(os.path.join(incomingPath, fileName))
+    with open(path) as f:
+        return f.readline()
+
+def convertTime(input):
+    input = input.split("+")[0] # time zone parsing does not work correctly in the jython version used here
+    objDate = datetime.strptime(input, '%Y-%m-%dT%H:%M:%S.%f')
+    return datetime.strftime(objDate,'%Y-%m-%d %H:%M:%S')
+
+def getTimeStamp():
+    now = datetime.now()
+    ts = str(now.minute)+str(now.second)+str(now.microsecond)
+    return ts
+
+def copyLogFilesTo(logFiles, filePath, targetFolderPath):
+    for logFile in logFiles:
+        src = os.path.join(filePath, logFile)
+        shutil.copy2(src, targetFolderPath)
+    copiedContent = os.listdir(targetFolderPath)
+    if len(copiedContent) != len(logFiles):
+        raise AssertionError("Not all log files have been copied successfully to target log folder.")
+
+def createLogFolder(targetPath):
+    ts = getTimeStamp()
+    newLogFolder = os.path.join(targetPath, os.path.join(ts, "logs"))
+    os.makedirs(newLogFolder)
+    return newLogFolder
+
+def createExperimentFromMeasurement(transaction, currentPath, space, project, measurement, origin, rawDataPerSample):
+    os.listdir(currentPath)
```

Comment:
Some more docs and I removed the `os.listdir(currentPath)` as the expression is not used
```suggestion
    """ Register the experiment with samples in openBIS.
    In order to register the Nanopore experiment with its measurements in openBIS, 
    we need to perform the following steps:
    
    1.) Create a new experiment in openBIS
    2.) Enrich it with metadata about the sequencing run (base caller, adapter, library kit, etc.)
    3.) Aggregate all log files into an own log folder per measurement
    4.) Create a new sample for every measurement
    
    The Map rawDataPerSample contains all DataFolders per sample code:
    
    [
       "QBiC sample id":
           [
            "fast5fail": DataFolder,
            "fast5pass": DataFolder,
            "fastqfail": DataFolder,
            "fastqpass": DataFolder
           ],
      "Other sample id":   // In case of pooled samples
         ...
    ]
    """
```